openfl/component/straggler_handling_functions/straggler_handling_function.py

```diff
@@ -5,15 +5,49 @@
 
 from abc import ABC
 from abc import abstractmethod
+from typing import Callable
 
 
 class StragglerHandlingFunction(ABC):
     """Federated Learning straggler handling interface."""
 
     @abstractmethod
-    def straggler_cutoff_check(self, **kwargs):
+    def start_policy(
+        self, callback: Callable, collaborator_name: str
+    ) -> None:
+        """
+        Start straggler handling policy for collaborator for a particular round.
+        NOTE: Refer CutoffTimeBasedStragglerHandling for reference.
+
+        Args:
+            callback: Callable
+                Callback function for when straggler_cutoff_time elapses
+            collaborator_name: str
+                Name of the collaborator
+
+        Returns:
+            None
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def reset_policy_for_round(self) -> None:
+        """
+        Control whether to start the timer or not.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def straggler_cutoff_check(self, **kwargs) -> bool:
```

Comment:
Right now it's not clear what kind of inputs this method expects. Why not define it a bit more explicitly, as in:
```python
def straggler_cutoff_check(self, num_collaborators_done, all_collaborators) -> bool:
```
or even
```python
def straggler_cutoff_check(self, num_collaborators_done, num_all_collaborators) -> bool:
```

Reply:
Method definition is changed to "def straggler_cutoff_check(self, num_collaborators_done: int, num_all_collaborators: int, **kwargs) -> bool".

Reply:
Thanks, this is much better 😊